packages/venia-ui/lib/components/Wishlist/GalleryButton/galleryButton.js

```diff
@@ -0,0 +1,67 @@
+import React, { Fragment, useEffect } from 'react';
+import { AlertCircle, Check, Heart, Info } from 'react-feather';
+import { useIntl } from 'react-intl';
+import { useToasts } from '@magento/peregrine';
+import mergeClasses from '@magento/peregrine/lib/util/shallowMerge';
+import { useGalleryButton } from '@magento/peregrine/lib/talons/Wishlist/GalleryButton/useGalleryButton';
+
+import Icon from '../../Icon';
+import WishlistDialog from '../WishlistDialog';
+import defaultClasses from './galleryButton.css';
+
+const CheckIcon = <Icon size={20} src={Check} />;
+const ErrorIcon = <Icon size={20} src={AlertCircle} />;
+const HeartIcon = <Icon size={20} src={Heart} />;
+const InfoIcon = <Icon size={20} src={Info} />;
+
+const GalleryButton = props => {
+    const talonProps = useGalleryButton(props);
+    const {
+        buttonProps,
+        errorToastProps,
+        isSelected,
+        loginToastProps,
+        modalProps,
+        successToastProps
+    } = talonProps;
+
+    const classes = mergeClasses(defaultClasses, props.classes);
+
+    const { formatMessage } = useIntl();
+    const [, { addToast }] = useToasts();
+
+    useEffect(() => {
+        if (loginToastProps) {
+            addToast({ ...loginToastProps, icon: InfoIcon });
+        }
+    }, [addToast, formatMessage, loginToastProps]);
```

Comment:
```suggestion
    }, [addToast, loginToastProps]);
```

Do we need `formatMessage` in the dependency array?